source/common/quic/envoy_quic_server_stream.cc

```diff
@@ -65,6 +65,18 @@ EnvoyQuicServerStream::EnvoyQuicServerStream(
           stats, http3_options),
       headers_with_underscores_action_(headers_with_underscores_action) {}
 
+EnvoyQuicServerStream::~EnvoyQuicServerStream() {
+  // Only the downstream stream should clear the downstream of the
+  // memory account.
+  //
+  // There are cases where a corresponding upstream stream dtor might
+  // be called, but the downstream stream isn't going to terminate soon
+  // such as StreamDecoderFilterCallbacks::recreateStream().
+  if (buffer_memory_account_) {
+    buffer_memory_account_->clearDownstream();
```

Comment:
See comment in the H2 codec, it's not clear to me if this happens early enough.

Reply:
Done.